Firestore/core/src/remote/bloom_filter.cc

```diff
@@ -115,8 +115,30 @@ bool BloomFilter::MightContain(absl::string_view value) const {
 
 bool operator==(const BloomFilter& lhs, const BloomFilter& rhs) {
   return lhs.bit_count() == rhs.bit_count() &&
-         lhs.hash_count() == rhs.hash_count() &&
-         equal(lhs.bitmap().begin(), lhs.bitmap().end(), rhs.bitmap().begin());
+         lhs.hash_count() == rhs.hash_count() && CompareBits(lhs, rhs);
+}
+
+bool operator!=(const BloomFilter& lhs, const BloomFilter& rhs) {
+  return !(lhs == rhs);
+}
+
+bool CompareBits(const BloomFilter& lhs, const BloomFilter& rhs) {
```

Comment:
Consider simplifying your CompareBits() function to do this:

1. Compare the bit_counts(), returning false if they are different.
2. Compare each _byte_ (each element) of the vectors _except_ for the last byte, returning false if any are different.
3. Compare the _last_ byte of the vectors, zeroing out the padding bits, returning false if they are different.
4. Return true

What do you think?